crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs

```diff
@@ -175,7 +235,16 @@ public class OceanRenderer : MonoBehaviour
         readonly int sp_lodAlphaBlackPointFade = Shader.PropertyToID("_CrestLodAlphaBlackPointFade");
         readonly int sp_lodAlphaBlackPointWhitePointFade = Shader.PropertyToID("_CrestLodAlphaBlackPointWhitePointFade");
 
-        void Awake()
+#if UNITY_EDITOR
+        static float _lastUpdateEditorTime = -1f;
+        public static float LastUpdateEditorTime => _lastUpdateEditorTime;
+        static int _editorFrames = 0;
+#endif
+
+        BuildCommandBuffer _commandbufferBuilder;
+
+        // Drive state from OnEnable and OnDisable? OnEnable on RegisterLodDataInput seems to get called on script reload
+        void OnEnable()
```

Comment:
I will do this. should we never update in prefab scenes? i didnt think about it too deeply before now..

Reply:
Hm just bailing out puts the ocean in a weird state, i had to add a bunch of null ref guards, and after that the ocean doesnt recover fully when going into prefab editing and back out. I maybe have to detect this case and trigger a rebuild of the system :/.

What were you seeing that triggered this comment? Maybe its easier to try to get the ocean running in prefab scenes.

Reply:
That's strange because that's what I get without this change. If I go into prefab mode the ocean is has no lighting (expected) and doesn't recover when leaving. I get a bunch of null exceptions.

I'll check this out with latest changes to see if something has changed..

Reply:
I get this error when exiting MainScene prefab:

```
NullReferenceException: Object reference not set to an instance of an object
Crest.OceanChunkRenderer.ExpandBoundsForDisplacements (UnityEngine.Transform transform, UnityEngine.Bounds& bounds) (at Assets/Crest/Crest/Scripts/OceanChunkRenderer.cs:169)
Crest.OceanChunkRenderer.UpdateMeshBounds () (at Assets/Crest/Crest/Scripts/OceanChunkRenderer.cs:58)
Crest.OceanChunkRenderer.Update () (at Assets/Crest/Crest/Scripts/OceanChunkRenderer.cs:52)
```

If I go into MainSceneCore from MainScene the scene camera gets mess up. Its position is far away from the island and the near clip plane is really large.

Reply:
Oh I didn't try prefab scene before

Reply:
I was bailing out right at the top of both OnEnable and OnDisable. Having it work in prefab scenes would be better, but I thought this would be a quicker solution.

Reply:
Bailing out at top of both seems to work perfectly!? I cant say I have a full grasp why though. I guess it just stops the code from messing with the state when entering a prefab scene, and when it comes back it just picks up where it left off?

I'll push this as a proposal.

Reply:
Pretty much. From what I gathered, the actual scene appears to continue to run in the background (or at least the registered editor update function). So before we were creating a new ocean in the prefab scene, and I think the new one wasn't being destroyed. I remember seeing a duplicate at one point.

And yeah, when we come back OnEnable or OnDisable on the original ocean are not called so that supports your statement.